### ISSUE TITLE
[HOLD for payment 2024-05-09] [HOLD for payment 2024-05-08] Thread - Chat flickers after clicking on link in thread header

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#41307]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.68-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4524207
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Open the app and log in
2. Open a report with a long history
3. Send a message and start a thread or assign a task
4. Click on the link in the header to return to the chat

## Expected Result:
There should be no flickering when returning to the chat

## Actual Result:
The chat flickers when user is returned from thread

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/ea8dd444-0ccf-45da-8ac5-90e5d0e285b6

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @isabelastisser</details>

**Comment by izarutskaya** [comment]
@isabelastisser I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
We think this issue might be related to the #vip-vsb.

**Comment by francoisl** [comment]
I think this comes from https://github.com/Expensify/App/pull/40624, reverting that PR locally fixes the issue.
cc @bernhardoj @abdulrahuman5196 if you have an idea how to resolve without reverting.

**Comment by francoisl** [comment]
Going to go ahead with a revert for now, let's look into a long-term solution down the line.

**Comment by abdulrahuman5196** [comment]
> Going to go ahead with a revert to unblock the deploy for now, let's look into a long-term solution down the line.

Thank you for taking care of revert. We will circle back on the GH to fix the regression issue. 
cc: @bernhardoj

**Comment by isabelastisser** [comment]
@francoisl, can I close this issue, or do you need to complete the BZ list above first?

**Comment by francoisl** [comment]
All good, there's already a follow-up PR to address the regression, let's close.
